paddle/fluid/framework/ir/graph_pattern_detector.cc

```diff
@@ -2176,6 +2180,36 @@ PDNode *patterns::FirstBfloat16Ops::operator()() {
   return op;
 }
 
+PDNode *patterns::DuplicatedInputs::operator()() {
```

Comment:
what does it mean duplicated input and why only sum and concat (not elementwise_add, elementwise_mul ......) ?

Reply:
This is because concat and sum have an input vector that is marked  `.AsDuplicable()`
https://github.com/PaddlePaddle/Paddle/blob/0ed80e09fcbac3d62e35dc07fa451ce1a32d4eb3/paddle/fluid/operators/concat_op.cc#L109
At the graph_pattern_detecotr level, it's hard to find the right connection to add reorders before each concat and sum. That is why such a workaround was used.